tcs-service/src/main/resources/db/migration/schema/V4.21__refactor_programmeMembership.sql

```diff
@@ -0,0 +1,49 @@
+DROP TABLE `ProgrammeMembership`;
+CREATE TABLE `ProgrammeMembership` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `programmeMembershipType` varchar(255) DEFAULT NULL,
+  `programmeStartDate` date DEFAULT NULL,
+  `programmeEndDate` date DEFAULT NULL,
+  `periodOfGrace` int(11) DEFAULT NULL,
+  `programmeId` bigint(20) NOT NULL,
+  `trainingNumberId` bigint(20) DEFAULT NULL,
+  `personId` bigint(20) NOT NULL,
+  `rotation` varchar(255) DEFAULT NULL,
+  `rotationId` bigint(20) DEFAULT NULL,
+  `trainingPathway` varchar(255) DEFAULT NULL,
+  `amendedDate` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
+  PRIMARY KEY (`id`),
+  KEY `fk_ProgrammeMembership_person_id` (`personId`),
+  KEY `fk_ProgrammeMembership_programme_id` (`programmeId`),
+  KEY `fk_ProgrammeMembership_training_number_id` (`trainingNumberId`),
+  KEY `fk_ProgrammeMembership_rotation_id` (`rotationId`),
+  CONSTRAINT `fk_ProgrammeMembership_person_id` FOREIGN KEY (`personId`) REFERENCES `Person` (`id`),
+  CONSTRAINT `fk_ProgrammeMembership_programme_id` FOREIGN KEY (`programmeId`) REFERENCES `Programme` (`id`),
+  CONSTRAINT `fk_ProgrammeMembership_training_number_id` FOREIGN KEY (`trainingNumberId`) REFERENCES `TrainingNumber` (`id`),
+  CONSTRAINT `fk_ProgrammeMembership_rotation_id` FOREIGN KEY (`rotationId`) REFERENCES `Rotation` (`id`)
+) ENGINE=InnoDB AUTO_INCREMENT=0 DEFAULT CHARSET=utf8;
+
+INSERT INTO ProgrammeMembership (personId, programmeId, rotationId, rotation, programmeStartDate, programmeEndDate,
+programmeMembershipType, trainingPathway, periodOfGrace)
+SELECT
+personId, programmeId, rotationId, max(rotation), programmeStartDate, programmeEndDate,
```

Comment:
max(rotation) is necessary because within some record sets that collapse to a single ProgrammeMemberhip record there are null and actual values. The other max(...) fields should be all the same (i.e. max(x) = x for all records in the set) but there is an issue with periodOfGrace for a couple of records.

Reply:
Since last time we [cleaned up the rotations](https://github.com/Health-Education-England/TIS-TCS/pull/822) in January, there have been new null values happening. They are here: https://build.tis.nhs.uk/metabase/question/433. By checking the amendedDate, we can confirm they're new ones. 
I agree we can use max(...) to get the actual values and only when we refactor the logic to apply new ProgrammeMembership-CurriculumMembership structure, the null values won't happen any more.